src/deepmol/models/_utils.py

```diff
@@ -114,3 +115,61 @@ def _save_keras_model(file_path: str,
     save_to_disk(parameters_to_save, file_path_parameters)
 
 
+def multi_label_binarize(y_pred_proba, threshold=0.5) -> np.ndarray:
+    """
+    Binarize the predicted probabilities for multi-label classification.
+
+    Parameters
+    ----------
+    y_pred_proba: np.ndarray
+        Predicted probabilities with shape (n_samples, n_classes).
+    threshold: float
+        Threshold for binarization.
+
+    Returns
+    -------
+        np.ndarray: Binary predictions with shape (n_samples, n_classes).
+    """
+    n_samples, n_classes = y_pred_proba.shape
+    y_pred = np.zeros((n_samples, n_classes))
+    for i in range(n_classes):
+        y_pred[:, i] = (y_pred_proba[:, i] >= threshold).astype(int)
+    return y_pred
+
+
+def get_prediction_from_proba(dataset: Dataset, y_pred_proba: np.ndarray) -> np.ndarray:
+    """
+    Get predictions from predicted probabilities.
+
+    Parameters
+    ----------
+    dataset: Dataset
+        The dataset.
+    y_pred_proba: np.ndarray
+        Predicted probabilities with shape (n_samples, n_classes).
+
+    Returns
+    -------
+    np.ndarray: Predictions with shape (n_samples,).
+
+    """
+    if dataset.mode == "classification" or dataset.mode == "multitask":
```

Comment:
When dataset.mode = "multitask" we can have multitask classification and regression or both.

Reply:
maybe in multitask problems is better to have a list like ['classification', 'regression', ...]. That would affect other parts of the code, but maybe is best to have it that way. What do you think?

Reply:
done